porcelain/payments.go

```diff
@@ -185,6 +185,60 @@ func CreatePayments(ctx context.Context, plumbing cpPlumbing, config CreatePayme
 	return response, nil
 }
 
+// ValidateStoragePaymentCondition validates that condition of a voucher created for a storage payment meets expectations
+func ValidateStoragePaymentCondition(ctx context.Context, condition *types.Predicate, minerAddr address.Address, commP types.CommP, pieceSize *types.BytesAmount) error {
+	// a nil condition is always valid
+	if condition == nil {
+		return nil
+	}
+
+	if condition.To != minerAddr {
+		return errors.New("voucher condition addressed to wrong address")
+	}
+
+	if condition.Method != verifyPieceInclusionMethod {
+		return errors.Errorf("payment condition method, %s, should be %s", condition.Method, verifyPieceInclusionMethod)
+	}
+
+	if condition.Params == nil || len(condition.Params) != 2 {
+		return errors.New("payment condition does not contain exactly 2 parameters")
+	}
+
+	var clientCommP types.CommP
+	// lack of type data in params means commP isn't necessarily cbor decoded
+	clientCommPBytes, ok := condition.Params[0].([]byte)
+	if ok {
+		copy(clientCommP[:], clientCommPBytes)
+	} else {
+		clientCommP, ok = condition.Params[0].(types.CommP)
+		if !ok {
+			return errors.New("piece commitment is not a CommP")
+		}
+	}
```

Comment:
Surely this should be exactly one or the other. We're writing the code at both ends. Even if it's necessarily opaque, can we pick one?